ac_dc/deduplicate.py

```diff
@@ -423,7 +425,7 @@ def process(record):
             ds[s].map(process, num_proc=num_proc)
 
     flags = defaultdict(lambda: False)
-    for x, y in edges:
+    for x, y in enumerate(edges):
```

Comment:
Is edges a list of hashables ( int or tuples) since you are using enumerate? X will start from 0 to N right?

Reply:
The changes look good to me except with a few questions. Have you tested for memory? So glad it's faster. 

Is memory usage ok with large index? 

Really great work!

Reply:
It should have been `tqdm` instead of `enumerate` here. I will fix it.

Yes, I was able to run `find_duplicates` for a 20 GB (20 shards) dataset on a 32 GB machine. The peak memory usage is less than 15 GB.